erpnext/agriculture/doctype/plant_batch/plant_batch.js

```diff
@@ -46,8 +50,41 @@ frappe.ui.form.on('Plant Batch', {
 				});
 			}, __("Create"));
 		}
-
-	}
+	},
+	split_plant_batch: function(frm) {
+		frappe.prompt([{
+			fieldname: 'split_count',
+			label: __('New Plant Batch Untracked Count'),
+			fieldtype: 'Int',
+		},
+		{
+			fieldname: 'new_plant_batch_id',
+			label: __('New Plant Batch ID'),
+			fieldtype: 'Data',
+			reqd: 1
+		}],
+		(data) => {
+			frappe.call({
+				method: 'erpnext.agriculture.doctype.plant_batch.plant_batch.split_plant_batch',
+				args: {
+					untracked_count: frm.doc.untracked_count,
+					strain: frm.doc.strain,
+					start_date: frm.doc.start_date,
+					location: frm.doc.location,
+					split_count: data.split_count,
+					new_plant_batch_id: data.new_plant_batch_id,
+					reference_name: frm.doc.name
```

Comment:
Instead of passing each document value individually, just pass the entire document and use that:

```suggestion
					doc: frm.doc
```

You could also turn this into a class function in the server controller, and use `frm.call` to process it there